### PR TITLE
Expose the specific engine client

### PR DESCRIPTION
**Pull request by simoneNEMO**
Quick solution to #3 

A couple of thoughts:
- It is a bit suboptimal that the client getter return `any` type. Ideally with a bit of Typescript voodoo one could make so that `MagnifyManager.engine` returns the type of the specific engine instead of the abstract `MagnifyEngine` and this should solve the problem with the getter type as well (also the client getter does not need to be part of the abstract `MagnifyEngine`)
- The `magnify` service was not exported. Should we also change the filename from `magnify.ts` to `main.ts` as in most of other AdonisJS packages

### CODE REVIEW

File: tests/units/magnify_manager.spec.ts [hunk 1]
```diff
@@ -39,4 +42,31 @@ test.group('Magnify manager', () => {
...
+          new TypesenseEngine({
+            nodes: [{ url: 'http://localhost' }],
+            collectionSettings: {},
+            apiKey: '',
+          }),
+      },
+    })
+
+    assert.instanceOf(manager.engine('meilisearch').client, MeiliSearch)
+    // assert.instanceOf(manager.engine('algolia').client, Algoliasearch) // algolia exports a type and not a class
```
Comment (kerwanp) [inline tests/units/magnify_manager.spec.ts]: I think this package deserve to be well typed. We could use https://japa.dev/docs/plugins/expect-type to assert the types.

File: package.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
     ".": "./build/index.js",
     "./commands": "./build/commands/main.js",
     "./magnify_provider": "./build/providers/magnify_provider.js",
+    "./services/magnify": "./build/services/magnify.js",
```
Comment (kerwanp) [inline package.json]: :+1:

**Review by kerwanp — CHANGES_REQUESTED** [review]

**Comment by kerwanp** [comment]
Thanks a lot for the contribution!

> It is a bit suboptimal that the client getter return any type. Ideally with a bit of Typescript voodoo one could make so that MagnifyManager.engine returns the type of the specific engine

 I think you are right, this package deserve to be well typed. If you want to do it in this pull request, feel free to do it! If not I'm ok to merge as it is and I will work on it later.
Also, I think we could use https://japa.dev/docs/plugins/expect-type in the tests to ensure types are working correctly.

> The magnify service was not exported. Should we also change the filename from magnify.ts to main.ts as in most of other AdonisJS packages

Definitely!


Thanks a lot for the typos aswell.

**Comment by simoneNEMO** [comment]
Switched to a Map for the engines cache (which was actually never used before) and improved the `MagnifyManager.engine` return typing. Also tests have been updated accordingly.

**Comment by kerwanp** [comment]
All tests passed !

(I should really find a way to have the tests run properly in the CI)

image

**Pull request merged**